squashfs-mount.c

```diff
@@ -120,8 +133,23 @@ int main(int argc, char **argv) {
   if (mnt_context_set_target(cxt, mountpoint) != 0)
     exit_with_error("Failed to set target\n");
 
-  if (mnt_context_mount(cxt) != 0)
+  if (flock(squashfs_fd, LOCK_EX) != 0)
```

Comment:
Will this approach serialize across all users of a file, not just all instances of an individual user?
i.e. multiple users trying to launch a job that accesses `/scratch/e1000/hstoppel/store.squashfs` will all be serialized?

Reply:
yes

Reply:
> we could have scenarios where multiple users make thousands of competing attempts to acquire the lock

The libmount call takes about 1.5ms on my desktop, I doubt it'll be thousands of competing attempts in such a short interval

> Does Sarus attempt to solve this issue

They didn't run into this issue yet, but it's likely they would, since the code for mount(8) code looks similar: https://github.com/util-linux/util-linux/blob/11ebc3d26e64e766d2a2ec670ca1a4ad4afdc381/sys-utils/mount.c#L1038.

Reply:
util-linux 2.38 solved some race condition about reuse of loop devices, and there's some discussion, maybe relevant: https://www.spinics.net/lists/util-linux-ng/msg17033.html given that the error is `overlapping loop device exists for ...`

Reply:
If the locking is only at the node level, then I can't see so much competition for locks.